server.js

```diff
@@ -151,6 +152,66 @@ camp.notfound(/.*/, function(query, match, end, request) {
 
 // Vendors.
 
+/**
+ * Registers a new service in the shields.io system. `serviceClass` must be a
+ * class that extends `BaseService`.
+ */
+function registerService(serviceClass) {
+  // Regular expressions treat "/" specially, so we need to escape them
+  const escapedPath = serviceClass.uri.format.replace(/\//g, '\\/');
+  const fullRegex = '^' + escapedPath + '.(svg|png|gif|jpg|json)$';
+
+  camp.route(new RegExp(fullRegex),
+  cache(async (data, match, sendBadge, request) => {
+    // Assumes the final capture group is the extension
+    const format = match.pop();
+    const badgeData = getBadgeData(
+      serviceClass.category,
+      Object.assign({}, serviceClass.defaultBadgeData, data)
+    );
+
+    try {
+      const namedParams = {};
+      if (serviceClass.uri.capture.length !== match.length - 1) {
+        throw new Error(
+          `Incorrect number of capture groups (expected `+
+          `${serviceClass.uri.capture.length}, got ${match.length - 1})`
+        );
+      }
```

Comment:
I should likely just do this sanity check on the examples once during registration, rather than on every request. I think it'd provide a similar level of safety.